inst/include/common/model.hpp

```diff
@@ -17,19 +17,17 @@
 #define FIMS_COMMON_MODEL_HPP
 
 #include <memory>
+#include <future>
 
-#include "../population_dynamics/fleet/fleet.hpp"
-#include "../population_dynamics/population/population.hpp"
-#include "../population_dynamics/recruitment/recruitment.hpp"
-#include "../population_dynamics/selectivity/selectivity.hpp"
+#include "information.hpp"
 
 namespace fims {
 
 /**
  * Model base
  */
 template <typename T>
-class Model {
+class Model {//may need singleton
  public:
   static std::shared_ptr<Model<T> >
```

Comment:
This shared_ptr appears to be setup for a single model as what is being pointed to.
So the question I would like to ask is, if we have a set of FIMS models to be fitted
with different setups as an ensemble model, is this sufficiently general to handle this?

Reply:
I think this is a good point, but something that should be addressed in future milestones. I think refactoring can be done to consider ensemble models